SC_redesign/Crop_and_Soil/tests_SC/test_heat_units.py

```diff
@@ -106,38 +89,22 @@ def test_check_absorb_heat_for_input_errors(use_alt, meant, min_t, max_t):
     with pytest.raises(ValueError):
         heat._check_absorb_heat_for_input_errors(meant, min_t, max_t)
 
-@pytest.mark.parametrize("air_min,air_max,crop_min,crop_max", [
-    (1, 1, 1, 1),
-    (0, 0, 0, 0),
-    (0, 1, 0, 1),
-    (0, 3, 1, 2),
-    (1, 2, 0, 3),
-    # arbitrary
-    (18.25, 33.4, 15.0, 35.0),  # air within crop
-    (12.31, 33.4, 15.0, 35.0),  # air_min < crop_min
-    (18.25, 40.7, 15.0, 35.0),  # air_max > crop_max
-    (12.31, 40.7, 15.0, 35.0),  # crop within air
-])
-def test_determine_heat_unit_temperature(air_min, air_max, crop_min, crop_max):
-    """check that heat unit temperature sets variables correctly"""
-    obs_min = calc_minimum_heat_unit_temperature(air_min, crop_min)
-    obs_max = calc_maximum_heat_unit_temperature(air_max, crop_max)
-    heat = init_heat(minimum_temperature=crop_min, maximum_temperature=crop_max)
-    heat.determine_heat_unit_temperature(air_min, air_max)
-    assert heat.minimum_heat_unit_temperature == obs_min
-    assert heat.maximum_heat_unit_temperature == obs_max
-    assert heat.heat_unit_temperature == ((obs_min + obs_max) / 2)
 
 @pytest.mark.parametrize("temp", [0, 20.5, None])
 def test_accumulate_heat_units(temp, mocker: MockerFixture):
     """check that accumulate_heat_units() calls the right functions"""
     patch_a = mocker.patch("SC_redesign.Crop_and_Soil.crop.heat_units.HeatUnits.assign_new_heat_units")
     patch_b = mocker.patch("SC_redesign.Crop_and_Soil.crop.heat_units.HeatUnits.add_heat_units")
     heat = init_heat()
+    expect = init_heat()
     heat.accumulate_heat_units(temp)
+    expect.assign_new_heat_units(temp)
+    expect.add_heat_units()
+    assert heat.accumulated_heat_units == expect.accumulated_heat_units
     assert patch_a.assert_called_once
     assert patch_b.assert_called_once
```

Comment:
Good thinking for how to solve this test problem. You don't need to change anything, but for the future, I think it would've been sufficient to do:

```suggestion
def test_accumulate_heat_units(temp, mocker: MockerFixture):
    """check that accumulate_heat_units() calls the right functions"""
    patch_a = mocker.patch("SC_redesign.Crop_and_Soil.crop.heat_units.HeatUnits.assign_new_heat_units")
    patch_b = mocker.patch("SC_redesign.Crop_and_Soil.crop.heat_units.HeatUnits.add_heat_units")
    heat = init_heat()
    heat.accumulate_heat_units(temp)
    assert patch_a.assert_called_once
    assert patch_b.assert_called_once
```

since `assign_new_heat_units()` and `add_heat_units()` are tested elsewhere. Over-testing is always better than under-testing in my view, though. So, good job!

Reply:
Thanks Clay, I will keep that in mind when looking at future tests